src/NHibernate/UserTypes/IEnhancedUserType.cs

```diff
@@ -7,9 +9,14 @@ namespace NHibernate.UserTypes
 	public interface IEnhancedUserType : IUserType
 	{
 		/// <summary>
-		/// Parse a string representation of this value, as it appears
-		/// in an XML document.
+		/// Parse a string representation of this value.
 		/// </summary>
+		// 6.0 TODO: rename "FromString(string value)".
+		// Since 5.2
+		[Obsolete("This method was not used for parsing xml strings, but instead used for parsing already de-encoded " +
+			"strings originating from xml (so indeed, plain text strings). It will be renamed \"FromString\" in a " +
+			"future version. Implement a \"object FromString(string value)\" without any xml de-coding, called by your" +
+			"\"object FromXMLString(string xml)\", in order to get ready for the rename.")]
 		object FromXMLString(string xml);
```

Comment:
The supplied `xml` string is never xml encoded, contrary to what we could expect with such naming and documentation. So I have considered a name change is required there. I do not see a better pattern for renaming an interface method than the one showcased here.